### PR TITLE
Accept primary compute spec in /promote, promotion corner cases testing

### PR DESCRIPTION
**Pull request by myrrc**
https://github.com/neondatabase/cloud/issues/19011
- Accept `ComputeSpec` in `/promote` instead of just passing safekeepers and LSN. Update API spec
- Add corner case tests for promotion when promotion or perwarm fails (using failpoints)
- Print root error for prewarm and promotion in status handlers

### CODE REVIEW

File: libs/compute_api/src/responses.rs [hunk 1]
```diff
@@ -90,11 +90,11 @@ pub enum PromoteState {
     Failed { error: String },
 }
 
-#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+#[derive(Deserialize, Default, Debug)]
 #[serde(rename_all = "snake_case")]
-/// Result of /safekeepers_lsn
-pub struct SafekeepersLsn {
-    pub safekeepers: String,
+pub struct PromoteConfig {
```
[the remark this answers is not in the document]
Comment (myrrc) [reply libs/compute_api/src/responses.rs:95]: Fixed

File: compute_tools/src/http/openapi_spec.yaml [hunk 2]
```diff
@@ -591,18 +584,15 @@ components:
...
       type: object
       required:
-        - safekeepers
-        - wal_flush_lsn
+        - spec
       properties:
-        safekeepers:
-          description: Primary replica safekeepers
-          type: string
-        wal_flush_lsn:
```
Comment (mtyazici) [inline compute_tools/src/http/openapi_spec.yaml:603]: Does it not require wal_flush_lsn now? Or is that going to be in spec?

Comment (myrrc) [reply compute_tools/src/http/openapi_spec.yaml:603]: Forgot that, it's still required, I'll fix the spec

File: test_runner/regress/test_lfc_prewarm.py [hunk 3]
```diff
@@ -164,6 +164,24 @@ def test_lfc_prewarm(neon_simple_env: NeonEnv, method: PrewarmMethod):
     check_prewarmed(method, client, desired)
 
 
+@pytest.mark.skipif(not USE_LFC, reason="LFC is disabled, skipping")
+def test_lfc_prewarm_empty(neon_simple_env: NeonEnv):
```
[the remark this answers is not in the document]
Comment (myrrc) [reply test_runner/regress/test_lfc_prewarm.py:167]: Fixed
